### PR TITLE
Architecture Updates and Simple Implementations

### PR DESCRIPTION
**Pull request by david-boles**
Lots of stuff!

### CODE REVIEW

File: src/org/team5940/robot_core/modules/SimpleModule.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	 */
+	private final String name;
+	
+	/**
+	 * Stores the submodules of this Module.
+	 */
+	private final ModuleHashTable<Module> subModules;
+	
+	/**
+	 * Creates a new ProcedureModule with given arguments.
```
Comment (mbentley124) [inline src/org/team5940/robot_core/modules/SimpleModule.java]: Shouldn't it say just Module rather then ProcedureModule.

File: src/org/team5940/robot_core/modules/SimpleModule.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+public abstract class SimpleModule implements Module {
+	/**
+	 * Stores the name of this Module.
+	 */
+	private final String name;
+	
+	/**
+	 * Stores the submodules of this Module.
+	 */
+	private final ModuleHashTable<Module> subModules;
```
Comment (mbentley124) [inline src/org/team5940/robot_core/modules/SimpleModule.java:12]: Does this work as final if it is being set later.

Comment (david-boles) [reply src/org/team5940/robot_core/modules/SimpleModule.java:12]: It should, it's only set once...

Comment (mbentley124) [reply src/org/team5940/robot_core/modules/SimpleModule.java:12]: My bad. I didn't think that worked but you are right.

File: src/org/team5940/robot_core/modules/ownable/SimpleOwnableModule.java [hunk 3]
```diff
@@ -62,45 +41,23 @@ public synchronized boolean isOwnedBy(Thread t) {
...
-	 */
 	@Override
-	public synchronized boolean relinquishOwnershipFor(Thread t) throws IllegalArgumentException {
+	public synchronized void relinquishOwnershipFor(Thread t) throws IllegalArgumentException {
 		if (t==null){
 			throw new IllegalArgumentException();
 		}
-		if (this.isOwnedBy(t) == true && t != null) {
+		if (this.isOwnedBy(t)) {
 			currentOwner = null;
```
Comment (mbentley124) [inline src/org/team5940/robot_core/modules/ownable/SimpleOwnableModule.java:59]: Don't you want to set the owner to t.

**Review by mbentley124 — APPROVED** [review]
Looks good

**Pull request merged**